spmd/tensor/device_mesh.py

```diff
@@ -261,18 +256,23 @@ def scatter(
             scatter_list (List[torch.Tensor]): list of tensors to scatter.
             mesh_dim (int, optional): indicate which mesh dimension we want
                 to scatter on, we by default choose the first rank on the
-                mesh dimension.
+                mesh dimension as source of truth.
 
         Returns:
             A :class:`torch.Tensor` object
         """
+        my_coordinate = self.get_coordinate_on_dim(mesh_dim)
+        # TODO: what should happen if rank is not in the mesh?
```

Comment:
Should we just silently return? I think we had discussed this before. In general we think that a good principle is that no-op for non participating members is a good behavior.

Reply:
Yeah, i think the tricky part is that we need to return something legit, i.e. for uneven sharding, the return tensor shape should be somewhat match, let me try remove this in a follow up PR and add some tests.